common/src/main/java/com/microsoft/identity/common/internal/platform/IDevicePopManager.java

```diff
@@ -90,6 +95,23 @@
      */
     void getRequestConfirmation(TaskCompletedCallbackWithError<String, ClientException> callback);
 
+    /**
+     * Signs an arbitrary piece of String data.
+     *
+     * @param input The input to sign.
+     * @return The input data, signed by our private key.
+     */
+    String sign(String input) throws ClientException;
```

Comment:
TODO: Do we need to accept something like a `SigningParams`; for instance, to specify an algorithm?

Also, the OneAuth API assumes signing cannot fail... we need some way to return error if this operation fails

Reply:
Made the assumption that this is going to use `SHA256WithRSA` and implemented accordingly

Reply:
7b21d1045de1feebb8686df19dc5a7557b1fd16c